modular_skyrat/modules/trim_tokens/code/cards_ids.dm

```diff
@@ -0,0 +1,25 @@
+// Proc to use a trim token on an ID
+/obj/item/card/id/proc/apply_token(obj/item/trim_token/token, mob/user)
+	if(token.has_required_trim)
+		if(!(src.trim in token.valid_trims))
+			to_chat(user, "<span class='warning'>The trim of your ID is not valid for this trim token.</span>")
+			return
+	if(token.uses == 0)
+		to_chat(user, "<span class='warning'>The [token.name] disintegrates as you press it against your ID, it probably only barely had enough charge left to keep its shape!</span>")
+		qdel(token)
+		return
+
+	// Just to make sure to give feedback if it requires a better card to grant the trim.
+	if(SSid_access.apply_trim_to_card(src, token.token_trim, copy_access = token.force_access))
+		playsound(src, token.usesound, 40)
+		to_chat(user, "<span class='notice'>The [token.name] fuses with your ID, replacing its trim with a [token.assignment] trim!</span>")
+		if(!(token.uses == INFINITE))
```

Comment:
Aaaaaa shitcode, infinite is a defined number. This will always return FALSE.
Do and if  else statement like: if(!token.uses) else